sound/smooth_swing_v2.h

```diff
@@ -94,38 +105,66 @@ class SmoothSwingV2 : public SaberBasePassThrough {
         if (speed < smooth_swing_config.SwingStrengthThreshold) {
 #if 1
           if (monitor.ShouldPrint(Monitoring::MonitorSwings)) {
-            STDOUT.print("speed: ");
-            STDOUT.println(speed);
+           STDOUT.print("speed: ");
+           STDOUT.println(speed);
           }
 #endif
           break;
         }
         state_ = SwingState::ON;
         
       case SwingState::ON:
+        //check for AccentSwingThreshold, presence of accent swings and if the accent player is stopped (this prevents clipping)
+        if (speed >=smooth_swing_config.AccentSwingSpeedThreshold
+          && accent_swings_present
+            && (A.player->isPlaying() || B.player->isPlaying())
+        {
+          //allocate player
+          if(!accent_player_) {
+            accent_player_ = GetFreeWavPlayer();
+          }
+          else {
+            if (!accent_player_->isPlaying()) {
+              accent_player_->PlayOnce(&swng);
+            }
+          }
+        }
         if (speed >= smooth_swing_config.SwingStrengthThreshold * 0.9) {
           float swing_strength =
-            std::min<float>(1.0, speed / smooth_swing_config.SwingSensitivity);
+          std::min<float>(1.0, speed / smooth_swing_config.SwingSensitivity);
           A.rotate(-speed * delta / 1000000.0);
           // If the current transition is done, switch A & B,
           // and set the next transition to be 180 degrees from the one
           // that is done.
           while (A.end() < 0.0) {
-            B.midpoint = A.midpoint + 180.0;
-	    Swap();
+           B.midpoint = A.midpoint + 180.0;
+           Swap();
           }
+          float accent_volume = 0.0;
           float mixab = 0.0;
           if (A.begin() < 0.0)
-            mixab = clamp(- A.begin() / A.width, 0.0, 1.0);
-
+           mixab = clamp(- A.begin() / A.width, 0.0, 1.0);
+          
           float mixhum =
             powf(swing_strength, smooth_swing_config.SwingSharpness);
-
+          
           hum_volume =
             1.0 - mixhum * smooth_swing_config.MaximumHumDucking / 100.0;
-
+          
           mixhum *= smooth_swing_config.MaxSwingVolume;
-
+          
+          if (on_) {
+            // We need to stop setting the volume when off, or playback may never stop.
+            A.set_volume(mixhum * mixab);
+            B.set_volume(mixhum * (1.0 - mixab));
+            //This volume will scale with swing speed but is modulated by AccentSwingVolumeSharpness.
+            if (accent_player_) {
+              accent_volume =
```

Comment:
need to check if accent player is playing.
If it is not, we don't need to calculate and set the volume.
In fact, if it is not, we should free it.